model/infinispan/src/main/java/org/keycloak/crls/infinispan/InfinispanCrlStorageProvider.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.crls.infinispan;
+
+import org.infinispan.Cache;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachelistener.annotation.*;
+import org.infinispan.notifications.cachelistener.event.*;
+import org.jboss.logging.Logger;
+import org.keycloak.crls.CrlEntry;
+import org.keycloak.crls.CrlLoader;
+import org.keycloak.crls.CrlStorageProvider;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.truststore.TruststoreProvider;
+import org.wildfly.security.auth.server.SecurityIdentity;
+
+import javax.security.auth.x500.X500Principal;
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.cert.*;
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Crl Storage Provider backed by Infinispan cache
+ *
+ * @author Joshua Smith
+ * @author Scott Tustison
+ */
+public class InfinispanCrlStorageProvider implements CrlStorageProvider {
+
+    private static final ExecutorService EXECUTOR_SERVICE = Executors.newCachedThreadPool();
```

Comment:
Is it possible to rather use `session.getProvider(ExecutorsProvider.class).getExecutor("crl")` instead of this static variable?